rskj-core/src/main/java/co/rsk/pcc/bto/DeriveExtendedPublicKey.java

```diff
@@ -0,0 +1,138 @@
+ *
+ * @author Ariel Mendelzon
+ */
+public class DeriveExtendedPublicKey extends NativeMethod {
+    private final CallTransaction.Function function = CallTransaction.Function.fromSignature(
+            "deriveExtendedPublicKey",
+            new String[]{"string", "string"},
+            new String[]{"string"}
+    );
+
+    private final HDWalletUtilsHelper helper;
+
+    private final static String XPUB_AND_PATH_NULL = "Must provide xpub and path arguments. None was provided.";
+    private final static String INVALID_XPUB = "Invalid extended public key '%s";
+
+    public DeriveExtendedPublicKey(ExecutionEnvironment executionEnvironment, HDWalletUtilsHelper helper) {
+        super(executionEnvironment);
+        this.helper = helper;
+    }
+
+    @Override
+    public CallTransaction.Function getFunction() {
+        return function;
+    }
+
+    @Override
+    public Object execute(Object[] arguments) {
+        if (arguments == null) {
+            throw new NativeContractIllegalArgumentException(XPUB_AND_PATH_NULL);
+        }
+        String xpub = (String) arguments[0];
+        String path = (String) arguments[1];
+
+        NetworkParameters params = helper.validateAndExtractNetworkFromExtendedPublicKey(xpub);
+        DeterministicKey key;
+        try {
+            key = DeterministicKey.deserializeB58(xpub, params);
+        } catch (IllegalArgumentException e) {
+            throw new NativeContractIllegalArgumentException(String.format(INVALID_XPUB, xpub), e);
+        }
+
+        // Path must be of the form S, with S ::= n || n/S with n an unsigned integer
+
+        // Covering special case: upon splitting a string, if the string ends with the delimiter, then
+        // there is no empty string as a last element. Make sure that the whole path starts and ends with a digit
+        // just in case.
+        if (path == null || path.length() == 0 || !isDecimal(path.charAt(0)) || !isDecimal(path.charAt(path.length()-1))) {
+            throwInvalidPath(path);
+        }
+
+        String[] pathChunks = path.split("/");
+
+        if (Arrays.stream(pathChunks).anyMatch(s -> !isDecimal(s))) {
+            throwInvalidPath(path);
+        }
+
+        List<ChildNumber> pathList;
+        try {
+            pathList = HDUtils.parsePath(path);
+        } catch (NumberFormatException ex) {
+            throw new NativeContractIllegalArgumentException(getInvalidPathErrorMessage(path), ex);
+        }
+
+        DeterministicKey derived = key;
+        for (ChildNumber pathItem : pathList) {
+            derived = HDKeyDerivation.deriveChildKey(derived, pathItem.getI());
+        }
+
+        return derived.serializePubB58(params);
+    }
+
+    @Override
+    public boolean isEnabled() {
+        return true;
+    }
+
+    @Override
+    public boolean onlyAllowsLocalCalls() {
+        return false;
+    }
+
+    @Override
+    public long getGas(Object[] parsedArguments, byte[] originalData) {
+        return 55_000L;
```

Comment:
Cost is fixed but it should be proportional to the number of `/` in the path. If 55k is worst case, then a maximum number of `/` must be added.